crate/config_utils/src/config_utils.rs

```diff
@@ -125,21 +114,21 @@ pub fn location(
 }
 
 pub trait ConfigUtils: Default {
-    fn to_toml(&self, conf_path: &PathBuf) -> Result<(), ConfigUtilsError>
+    fn to_toml(&self, conf_path: &str) -> Result<(), ConfigUtilsError>
     where
         Self: serde::ser::Serialize + std::fmt::Debug,
     {
         self.save(conf_path, false)
     }
 
-    fn to_json(&self, conf_path: &PathBuf) -> Result<(), ConfigUtilsError>
+    fn to_json(&self, conf_path: &str) -> Result<(), ConfigUtilsError>
     where
         Self: serde::ser::Serialize + std::fmt::Debug,
     {
         self.save(conf_path, true)
     }
 
-    fn save(&self, conf_path: &PathBuf, json: bool) -> Result<(), ConfigUtilsError>
+    fn save(&self, conf_path: &str, json: bool) -> Result<(), ConfigUtilsError>
```

Comment:
Maybe we can create a `FileType` (or `SupportedFile`) enum holding both `Json` and `Toml` variants.
By doing that, we will get rid of the not-so-clear intent of the json boolean and we can add more file variant later on without modifying the `save` signature.